### PR TITLE
feat: update button styles

### PR DESCRIPTION
**Pull request by thisguyscode**

### CODE REVIEW

**Comment by thisguyscode** [comment]
@jonasbuechel just FYI, still a draft though.

File: src/components/Button/Button.stories.tsx [hunk 1]
```diff
@@ -3,81 +3,88 @@
 import React from "react";
 import { Meta, Story } from "@storybook/react";
 import { Button, ButtonProps, ButtonRounding, ButtonSize, ButtonStyle, ButtonType } from "./Button";
-import { IconIcons, IconActions } from "@foundation/Icon/Generated";
+import { ButtonEmphasis } from ".";
```
[the remark this answers is not in the document]
Comment (thisguyscode) [reply src/components/Button/Button.stories.tsx:6]: Oh yeah I guess that was automatically added within my IDE when I wrote ButtonEmphasis elsewhere. I think it works for sure but is not the pattern we use generally.

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -107,27 +35,14 @@ const typesMap: Record<ButtonType, "button" | "submit" | "reset"> = {
     [ButtonType.Reset]: "reset",
 };
 
-export type ButtonProps = {
```
[the remark this answers is not in the document]
Comment (thisguyscode) [reply src/components/Button/Button.tsx:110]: I was thinking the same actually, that ButtonProps is more useful in the Button component. You can change that for sure.

**Comment by imoutaharik** [comment]
@thisguyscode So I find out the problem was with the concatenation of tailwind classes so I reset the whole button classes file. We apparently may bypass the sonar cloud check so this can be merged.

File: src/foundation/Icon/Generated/IconDotsVertical.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React, { memo } from "react";
+import { GeneratedIconProps } from "@foundation/Icon/IconProps";
+import { IconSize, IconSizeMap } from "@foundation/Icon/IconSize";
+
+function IconDotsVertical(props: GeneratedIconProps): React.ReactElement<GeneratedIconProps> {
```
Comment (deniaz) [inline src/foundation/Icon/Generated/IconDotsVertical.tsx:5]: Can you open a separate PR for adding the Icon? It's not really relevant to the PR and will get lost in the CHANGELOG.

File: package.json [hunk 4]
```diff
@@ -140,9 +141,9 @@
         "@xstate/immer": "^0.3.0",
         "@xstate/react": "^1.6.3",
         "date-fns": "^2.28.0",
-        "react-datepicker": "^4.6.0",
         "framer-motion": "^6.2.1",
         "react-color": "^2.19.3",
+        "react-datepicker": "^4.6.0",
```
[the remark this answers is not in the document]
Comment (imoutaharik) [reply package.json:146]: No, it's not, I updated the Pr to last changes from main and this line somehow just changed place, it's not really an addition.

File: src/components/Button/ButtonTypes.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+export enum ButtonRounding {
+    Medium = "Medium",
+    Full = "Full",
+}
+export enum ButtonStyle {
```
[the remark this answers is not in the document]
Comment (thisguyscode) [reply src/components/Button/ButtonTypes.ts:7]: Did this so as to share it between both the main component file and the classes file.

**Comment by thisguyscode** [comment]
@imoutaharik can you resolve the SonarCloud thing here please? Also the commits miiiiight be unsigned, as with my new laptop I overlooked one setting. If so hopefully we can figure that one out.

**Comment by imoutaharik** [comment]
@thisguyscode I just reviewed the PR, thanks for adding the danger styles variations. I also fixed the code smell and recommitted the unsigned commits. Now, sonarCloud is still failing for the duplicated lines issues but we can ignore it for this specific case.

**Comment by thisguyscode** [comment]
@imoutaharik thanks! Are you able to solve the SonarCloud smell and merge, or is there some documentation on how to achieve this?

**Comment by imoutaharik** [comment]
> @imoutaharik thanks! Are you able to solve the SonarCloud smell and merge, or is there some documentation on how to achieve this?

The code smells are all fixed. The duplications can be ignored as we can't really make those tailwind classes reusable to a point that sonarcloud would be happier with it. The only thing missing here is another approval.

**Review by imoutaharik — APPROVED** [review]

**Review by jonasbuechel — APPROVED** [review]

**Comment by julianiff** [comment]
Closing: Moving to branch merged current with main (https://github.com/Frontify/fondue/pull/857) -> archive to keep merged changes

**Pull request closed without merging**